### ISSUE TITLE
Intermittent IOPub timeout problems on windows

### ISSUE THREAD

**Issue by akhmerov** [jupyter/nbclient#110]
When running windows CI, I observe sporadic execution failures with a warning

`WARNING | Timeout waiting for IOPub output`.

Related reports: https://github.com/mwouts/jupytext/issues/489 https://github.com/jupyter/jupyter-sphinx/issues/157
There was a WIP PR against nbconvert that seemingly should fix this (https://github.com/jupyter/nbconvert/pull/1183), but it was seemingly not finished nor taken over.

**Comment by SylvainCorlay** [comment]
cc @maartenbreddels since you authored the original PR.

**Comment by MSeal** [comment]
Are these test being run in parallel? https://github.com/jupyter/jupyter_client/issues/487 is a known race condition with port acquisition. I haven't had to time to implement the proposed handshake changes to remedy that problem.

If the original reported issue is still occurring on Windows, the PR referenced is now quite out of date as those methods have been made async so the solution likely needs to be rewritten. I am unlikely be to able to work on taking that over / confirming it's still the source problem in the near future personally.

**Comment by akhmerov** [comment]
No, the tests are run sequentially, at least in jupyter-sphinx.
